src/DotNet.Status.Web/Controllers/GitHubClientFactory.cs

```diff
@@ -21,7 +21,7 @@ public GitHubClientFactory(IOptions<GitHubClientOptions> githubClientOptions, IG
             _tokenProvider = tokenProvider;
         }
 
-        public async Task<IGitHubClient> CreateGitHubClientAsync(string owner, string repo)
+        public virtual async Task<IGitHubClient> CreateGitHubClientAsync(string owner, string repo)
```

Comment:
Is it ok to make this method virtual?

Reply:
Its the only method that does anything in this class.  If you make it virtual, the class becomes useless again.  And then you need to inject a bunch of options to let the constructor work, only to ignore them, which increases the complexity of the tests (having to configure a fake GitHubClientOptions and IGitHubTokenProvider) for no value gain.

It should get extracted into an interface if you want to mock something, and then mock the interface.

Reply:
@ChadNedzlek I am going to add this one also to the same issue I created yesterday. So I can work on it after I merge my PR. Is that fine?

Reply:
You're just making more work for yourself and introducing potential bugs.   It's a pretty small change, and I'd rather we don't check code we don't like into master, since it just increases the technical debt.  This change is so small that without it, this change introduces more debt than it fixes...

Reply:
Sure I will add this to the current PR.